File: backend/dataset/levelOrder/levelorder_5.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
      if not root:
        return []
      a = root
      l = [[root.val]]
      d = [a.left, a.right]
      while d:
        temp = []
        ll = []
        for i in d:
          if not i:
            continue
          ll += [i.val]
          temp += [i.left, i.right]
        else:
          if ll:
            l += [ll]
          d = temp
      return l

File: backend/CDataset/levelOrder/levelOrder_5.cpp
class Solution {
public:
    class TreeNode{
        public:
        TreeNode* left;
        TreeNode* right;
        int val;
    };

   vector<vector<int>> ans;
    queue<TreeNode*> q1;
    vector<vector<int>> levelOrder(TreeNode* root) {
        
        if(!root) return ans;
         int k;
         auto p=root;
         q1.push(p);
         TreeNode* t;
        
        while(q1.size())
        {   
            k=q1.size();vector<int> ca;
            while(k--)
            {
                 t=q1.front();q1.pop();
                 if(t->left)  q1.push(t->left);
                 if(t->right) q1.push(t->right);
                 ca.push_back(t->val); 
                           
            }
            ans.push_back(ca);
            }
            return ans;       
        }
};